go/pkg/client/bytestream_test.go

```diff
@@ -0,0 +1,294 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+
+	"google.golang.org/grpc"
+
+	bsgrpc "google.golang.org/genproto/googleapis/bytestream"
+	bspb "google.golang.org/genproto/googleapis/bytestream"
+)
+
+var (
+	logStreams    map[string]*logStream
+	logStreamData = []byte("Hello World! This is large data to send.")
+)
+
+type logStream struct {
+	logStreamID   string
+	logicalOffset int64
+	finalized     bool
+}
+
+func TestWriteBytesWithOffsetSuccess_LogStream(t *testing.T) {
+	tests := []struct {
+		description string
+		ls          *logStream
+		data        []byte
+		numIterate  int
+		opts        []ByteStreamWriteOption
+		wantBytes   int64
+	}{
+		{
+			description: "valid data with offset 0",
+			ls:          &logStream{logStreamID: "logstream1", logicalOffset: 0},
+			data:        logStreamData,
+			opts:        []ByteStreamWriteOption{ByteStreamOptOffset(0), ByteSteramOptFinishWrite(false)},
+			numIterate:  3,
+			wantBytes:   int64(len(logStreamData)),
+		},
+		{
+			description: "valid data with non-zero offset",
+			ls:          &logStream{logStreamID: "logstream2", logicalOffset: 4},
+			data:        logStreamData,
+			opts:        []ByteStreamWriteOption{ByteStreamOptOffset(4), ByteSteramOptFinishWrite(false)},
+			numIterate:  3,
+			wantBytes:   int64(len(logStreamData)),
+		},
+		{
+			description: "one big chunk",
+			ls:          &logStream{logStreamID: "logstream3", logicalOffset: 0},
+			data:        logStreamData,
+			opts:        []ByteStreamWriteOption{ByteStreamOptOffset(0), ByteSteramOptFinishWrite(true)},
+			numIterate:  1,
+			wantBytes:   int64(len(logStreamData)),
+		},
+		{
+			description: "empty data",
+			ls:          &logStream{logStreamID: "logstream4", logicalOffset: 0},
+			data:        []byte{},
+			opts:        []ByteStreamWriteOption{ByteStreamOptOffset(0), ByteSteramOptFinishWrite(false)},
+			numIterate:  1,
+			wantBytes:   0,
+		},
+	}
+
+	b := setup(t)
+	defer b.shutDown()
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			size := len(test.data)/test.numIterate + 1
+			var start int
+			start = int(test.opts[0].(ByteStreamOptOffset))
```

Comment:
Nit: Those two lines can be joined into one using a short declaration: `start := int(...)`.

Reply:
Done.